Orange/widgets/visualize/owscatterplot.py

```diff
@@ -571,12 +572,13 @@ def handleNewSignals(self):
         self.attr_box.setEnabled(True)
         self.vizrank.setEnabled(True)
         if self.attribute_selection_list and self.data is not None and \
-                self.data.domain is not None and \
-                all(attr in self.data.domain for attr
-                        in self.attribute_selection_list):
-            self.attr_x, self.attr_y = self.attribute_selection_list[:2]
+                self.data.domain is not None:
             self.attr_box.setEnabled(False)
             self.vizrank.setEnabled(False)
+            if all(attr in self.xy_model for attr in self.attribute_selection_list):
+                self.attr_x, self.attr_y = self.attribute_selection_list
+            else:
+                self.attr_x, self.attr_y = None, None
```

Comment:
Perhaps it would be better to set `attr_x` and `attr_y` to the first two variables in the model, as default? Or did I miss something?

Reply:
With this change, I wanted to explicitly show that the user provided the wrong features on the input (so nothing is plotted). Do you think it is better to plot the first two features and ignore the features input in this case?